### DISCUSSION TITLE
uxm_setup questions

### DISCUSSION THREAD

**Discussion by aashrita** [simonsobs/sodetlib#391] [Q&A] [answered]
Hi,

I had some questions about the code written in uxm_setup and what certain things mean. 

E.g. in uxm_setup, here is a snippet of code where I don't understand why certain methods are used/what they necessarily do:

```
    S.all_off()  # Turn off Flux ramp, tones, and biases
    S.set_rtm_arb_waveform_enable(0)
    S.set_filter_disable(0)
    S.set_downsample_factor(exp['downsample_factor'])
    if exp['coupling_mode'] == 'dc':
        S.set_mode_dc()
    else:
        S.set_mode_ac()

    for band in bands:
        S.set_synthesis_scale(band, exp['synthesis_scale'])
```

What does the rtm_arm_waveform_enable do? What filter is being disables? What is being downsampled & why? What is the synthesis scale?

Thanks!

**Comment by msilvafe** [comment, marked as answer]
What does the rtm_arm_waveform_enable do? 

This is what's used to play a waveform on the TES bias lines DACs it's used in bias steps and complex impedance at the moment in sodetlib.

What filter is being disables? 
This is the low pass filter applied before decimating. The data is acquired on the pcie at the flux ramp rate (typically 4KHz) and then we downsample it before righting to diskz Typically we write to disk at 200 Hz or a downsampling factor of 20 which requires a low pass filter that rolls off anything above 100 Hz).

What is being downsampled & why? 

What is above, why is because we 4kHz is more bandwidth than we need so we save a lot of space on disk by downsampling before writing to file.

What is the synthesis scale?

This is one of the parameters that determines the amplitude of a tone generated in the firmware but this overwrites the value set by your pysmurf config by the value in your device config.
